1.23 CLI Release Docs (#3388)
astronomer/docs: astro/cli/release-notes.md

- **Stable versions**: 1.21.0, 1.20.1, and 1.19.4. See [Astro CLI release and lifecycle policy](cli/release-lifecycle-policy.md).

## Astro CLI 1.23.0

Release date: February 14, 2024

### Changes to existing CLI command flags

> The following flags have been updated, but will continue to work with a deprecation notice until the next major release of the Astro CLI:

Reviewer: They will continue to work with a deprecation notice until this release is deprecated. They will stop working when  v1.25.0 is released

> The following flags have been updated, but will continue to work with a deprecation notice until the next major release of the Astro CLI:

Author: ```suggestion
The following flags have been updated, but will continue to work with a deprecation notice until the v1.25.0 release of the Astro CLI:
```

- `astro deployment logs --key-word` ia a new flag that allows you to search your Audit logs for an exact key word or phrase.
- `astro deployment create --cluster-type` is now `astro deployment create --type`.
- `astro deployment create --enforce-cicd` is now `astro deployment create --cicd-enforcement`.

### Kubernetes worker configurations are now consistent with the Astro Cloud UI

[Deployment files](deployment-file-reference.md) now include some new and updated fields for Deployment configuration to match the options available in the Cloud UI.

You now use the `default_task_pod_cpu`, `default_task_pod_memory`, and `default_worker_type` fields in a Deployment file to update your Kubernetes workers instead of creating or updating a Kubernetes default worker queue.

With this new functionality, you can't use the following commands to change your Kubernetes worker configuration if you've configured your Deployment file:

- `astro deployment create --from-file`
- `astro deployment update --from-file`
- `astro deployment worker-queue create`
- `astro deployment worker-queue update`

> Instead, you can use the new `--default-task-pod-cpu`, `--default-task-pod-memory`, `--resource-quota-cpu`, or `--resource-quota-memory` flags with [`astro deployment create`](astro-deployment-create.md) and [`astro deployment update`](astro-deployment-update.md) to edit your Kubernetes worker configuration.

Reviewer: you can also use new fields in the deployment file

### Changes to Deployment file configurations

The following changes have been made to the format of [Deployment files](deployment-file-reference.md):

- `cluster_name` is no longer used for standard Deployment files.
- `scheduler_size` is no longer case sensitive.
- Possible values for `cloud_provider` are now `gcp`, `aws`, and `azure`. This input is not case sensitive.

> - Possible values for `deployment_type` are now `standard`, `dedicated`, and `hybrid`. This input is not case sensitive

Reviewer: should we also indicate the old values of hosted_shared, hosted_dedicated, and hosted_standard still work?

> - Possible values for `deployment_type` are now `standard`, `dedicated`, and `hybrid`. This input is not case sensitive

Author: ```suggestion
- Possible values for `deployment_type` now include `standard`, `dedicated`, and `hybrid` in addition to the existing values of `hosted_shared`, `hosted_dedicated`, and `hosted_standard`. This input is not case sensitive
```

> - Possible values for for the `executor` field are now `celery` and `kubernetes`. `CeleryExecutor` and `KubernetesExecutor` still work, but they cause deprecation notices. This input is not case sensitive, so, for example, `celeryexecutor` still works.

Reviewer: There is not deprecation notice for `CeleryExecutor` and `KubernetesExecutor` inputs. Those will continue to work indefinitely for now

> - Possible values for for the `executor` field are now `celery` and `kubernetes`. `CeleryExecutor` and `KubernetesExecutor` still work, but they cause deprecation notices. This input is not case sensitive, so, for example, `celeryexecutor` still works.

Author: ```suggestion
- Possible values for for the `executor` field are now include `celery` and `kubernetes`. `CeleryExecutor` and `KubernetesExecutor` still work. This input is not case sensitive, so, for example, `celeryexecutor` still works.
```

- (_Astro Hosted only_) `default_task_pod_cpu`, `default_task_pod_memory`, `resource_quota_cpu`, and `resource_quota_memory` are new fields for Astro Hosted deployments.
- (_Astro Hybrid only_)`default_worker_type` is a new field for Hybrid deployments that use the Kubernetes executor.

### Additional improvements

- `astro deployment logs --key-word` is a new flag that allows you to search your audit logs for an exact key word or phrase.

> - The CLI no longer auto-selects the Deployments where your code deploys when you use it for Deployments that don't have CI/CD enforcement. Or, it also does not auto-select a Deployment if you have the `ASTRO_API_TOKEN`, `ASTRONOMER_KEY_ID`, or `ASTRONOMER_KEY_SECRET` environment variables set.

Reviewer: The auto-select doesn't change based on CI/CD enforcement.
Auto-select works this way now:
- by default auto-select is disabled. You will need to select a deployment if you are logged into the CLI as a user.
- If you used a token/key to login by setting one of the environment variables `ASTRO_API_TOKEN`, `ASTRONOMER_KEY_ID`, or `ASTRONOMER_KEY_SECRET` auto-select is enabled. This is important because it ensures that users with CI/CD scripts that rely on auto-select will continue to work
- There is a new config `auto_select`. If it is set to true in the config file like `auto_select: true` auto-select will always be enabled

Reviewer: Also we should probably be more clear on what auto-select is. Auto-select is where the CLI automatically selects a deployment if there is only one available to the user in the workspace that is currently selected as the CLIs context. auto-select will be turned off by default because it has caused users to deploy to the wrong deployment in the past(they didn't know what workspace they were in, used wrong deployment token, etc...)

- **Stable versions**: 1.21.0, 1.20.1, and 1.19.4. See [Astro CLI release and lifecycle policy](cli/release-lifecycle-policy.md).

## Astro CLI 1.23.0

Release date: February 14, 2024

### Changes to existing CLI command flags

> These flags have been updated, but will continue to work with a deprecation notice.

Reviewer 2: ```suggestion
The following flags have been updated, but will continue to work with a deprecation notice until the next major release of the Astro CLI:
```

> - `astro deployment logs --key-word` ia a new flag that allows you to search your Audit logs for an exact key word or phrase.
> - `astro deployment create --cluster-type` is now `astro deployment create --type`.
> - `astro deployment create --enforce-cicd` is now `astro deployment create --cicd-enforcement`.

Reviewer 2: ```suggestion
- `astro deployment logs --key-word` is a new flag that allows you to search your audit logs for an exact key word or phrase.
- `astro deployment create --cluster-type` is now `astro deployment create --type`.
- `astro deployment create --enforce-cicd` is now `astro deployment create --cicd-enforcement`.
```

Reviewer 2: The first one is sort of the odd one out here, should we put this under "additional improvements instead?

> ### New fields to update Kubernetes worker types, CPU, and memory

Reviewer 2: ```suggestion
### Kubernetes worker configurations are now consistent with the Astro Cloud UI
```

> If you use a [Deployment file](deployment-file-reference.md) to manage your Deployments as code, there are some new and changed fields.

Reviewer 2: ```suggestion
[Deployment files](deployment-file-reference.md) now include some new and updated fields for Deployment configuration to match the options available in the Cloud UI.
```

> You can now use the new `default_task_pod_cpu`, `default_task_pod_memory`, and `default_worker_type` fields in a Deployment file to update your Kubernetes workers instead of creating or updating a Kubernetes default worker queue.

Reviewer 2: ```suggestion
You now use the `default_task_pod_cpu`, `default_task_pod_memory`, and `default_worker_type` fields in a Deployment file to update your Kubernetes workers instead of creating or updating a Kubernetes default worker queue.
```

> With this new functionality, you can't use the following commands to change your Kubernetes worker configuration if you've configured your Deployment file:

Reviewer 2: ```suggestion
With this new functionality, you can no longer use the following commands to update a Deployment running the Kubernetes executor:
```
Does this need a breaking change warning?

Author: @sunkickr - remind me again, is this now "will work but with a deprecation warning" or "won't work at all?"

- `astro deployment create --from-file`

> - `astro deployment update --from-file`

Reviewer 2: Why would these no longer work if we just added fields to support KE configs?

Reviewer: `astro deployment create --from-file` and `astro deployment update --from-file` are affected by the change but will still work. Before you had to make you changes to the worker-queue field and now you have use the new fields of `default_worker_type`(For hybrid), `default_task_pod_cpu` and `default_task_pod_memory`(for hosted)

Reviewer 2: @lzdanski I think it's slightly misleading to include these commands, as the commands themselves haven't changed but rather the fields you pass to them. I'd use this release note to focus specifically on the worker queue commands and leave the Deployment file information to the latter note

- `astro deployment worker-queue create`
- `astro deployment worker-queue update`

Instead, you can use the new `--default-task-pod-cpu`, `--default-task-pod-memory`, `--resource-quota-cpu`, or `--resource-quota-memory` flags with [`astro deployment create`](astro-deployment-create.md) and [`astro deployment update`](astro-deployment-update.md) to edit your Kubernetes worker configuration.

> ### Changes to Deployment file configurations

Reviewer 2: Should we move the Deployment file information from the previous note into here?

> ### Changes to Deployment file configurations

Reviewer 2: ```suggestion
### Changes to Deployment file configurations

The following changes have been made to the format of [Deployment files](deployment-file-reference.md):
```

> - `cluster_name` is not used for standard deployments.
> - Inputs for `scheduler_size` are not case sensitive.
> - Inputs for `cloud_provider` are `gcp`, `aws`, and `azure`. This input is not case sensitive.
> - Inputs for `deployment_type` are now `STANDARD`, `DEDICATED`, and `HYBRID`. This input is not case sensitive
> - Inputs for the `executor` field are now `CELERY` and `KUBERNETES` are the new inputs for the executor field. `CeleryExecutor` and `KubernetesExecutor` still work, but cause deprecation notices. This input is not case sensitive, so, for example, `celeryexecutor` still works.

Reviewer 2: ```suggestion
- `cluster_name` is no longer used for standard Deployment files.
- `scheduler_size` is no longer case sensitive.
- Possible values for `cloud_provider` are now `gcp`, `aws`, and `azure`. This input is not case sensitive.
- Possible values for `deployment_type` are now `standard`, `dedicated`, and `hybrid`. This input is not case sensitive
- Possible values for for the `executor` field are now `celery` and `kubernetes`. `CeleryExecutor` and `KubernetesExecutor` still work, but they cause deprecation notices. This input is not case sensitive, so, for example, `celeryexecutor` still works.
```

> <HostedBadge/>

Reviewer 2: Idk if I love inline usage of the badges, it looks a little out of place amongst the bullet points

> - `default_task_pod_cpu`, `default_task_pod_memory`, `resource_quota_cpu`, and `resource_quota_memory` are new fields for Hosted deployments.

Reviewer 2: ```suggestion
- (_Astro Hosted only_) `default_task_pod_cpu`, `default_task_pod_memory`, `resource_quota_cpu`, and `resource_quota_memory` are new fields for Astro Hosted deployments.
```
We could even get rid of the parentheses and just rely on the statement itself

<HybridBadge/>
- `default_worker_type` is a new field for Hybrid deployments that use the Kubernetes executor.

> ### Other changes

Reviewer 2: ```suggestion
### Additional improvements
```

- `astro deployment logs --key-word` ia a new flag that allows you to search your Audit logs for an exact key word or phrase.
- `astro deployment create --cluster-type` is now `astro deployment create --type`.
- `astro deployment create --enforce-cicd` is now `astro deployment create --cicd-enforcement`.

### Kubernetes worker configurations are now consistent with the Astro Cloud UI

You can now use Deployment files or the Astro CLI to [create or update Kubernetes worker configurations](astro/deployment-resources.md#configure-kubernetes-pod-resources).

[Deployment files](deployment-file-reference.md) now include some new and updated fields for Deployment configuration to match the options available in the Cloud UI. This also allows you to create or update Kubernetes worker configurations directly, instead of requiring you to update the worker resources by changing the Kubernetes worker queue configuration.

You can now use the `default_task_pod_cpu`, `default_task_pod_memory`, `default_worker_type`, `resource_quota_cpu`, and `resource_quota_memory` fields in a Deployment file to update your Kubernetes workers instead of creating or updating a Kubernetes default worker queue.

> With this new functionality, the following commands to update a Deployment running the Kubernetes executor continue to work, but display a deprecation notice:

Reviewer 2: ```suggestion
With this new functionality, the following commands continue to work for Deployments running the Kubernetes executor, but they display a deprecation notice:
```
